src/coreclr/jit/emitriscv64.cpp

```diff
@@ -3503,51 +3513,76 @@ void emitter::emitDispInsName(
         }
         case 0x13:
         {
-            unsigned int opcode2 = (code >> 12) & 0x7;
-            const char*  rd      = RegNames[(code >> 7) & 0x1f];
-            const char*  rs1     = RegNames[(code >> 15) & 0x1f];
-            int          imm12   = (((int)code) >> 20); // & 0xfff;
-            // if (imm12 & 0x800)
-            //{
-            //    imm12 |= 0xfffff000;
-            //}
+            unsigned opcode2      = (code >> 12) & 0x7;
+            unsigned rd           = (code >> 7) & 0x1f;
+            unsigned rs1          = (code >> 15) & 0x1f;
+            int      imm12        = static_cast<int>(code) >> 20;
+            bool     isHex        = false;
+            bool     hasImmediate = true;
+            int      printLength  = 0;
+
             switch (opcode2)
             {
-                case 0x0: // ADDI
-                    printf("addi           %s, %s, %d\n", rd, rs1, imm12);
-                    return;
-                case 0x1:                                                         // SLLI
-                    printf("slli           %s, %s, %d\n", rd, rs1, imm12 & 0x3f); // 6 BITS for SHAMT in RISCV64
-                    return;
-                case 0x2: // SLTI
-                    printf("slti           %s, %s, %d\n", rd, rs1, imm12);
-                    return;
-                case 0x3: // SLTIU
-                    printf("sltiu          %s, %s, %d\n", rd, rs1, imm12);
-                    return;
-                case 0x4: // XORI
-                    printf("xori           %s, %s, 0x%x\n", rd, rs1, imm12);
-                    return;
-                case 0x5: // SRLI & SRAI
-                    if (((code >> 30) & 0x1) == 0)
+                case 0x0: // ADDI & MV & NOP
+                    if (imm12 != 0)
+                    {
+                        printLength = printf("addi");
+                    }
+                    else if ((rd != REG_ZERO) || (rs1 != REG_ZERO))
```

Comment:
Nit: What about like below? It seems comparing with `INS_nop` directly is more intuitive.
`if (code == INS_nop)
{
  printLength = printf("nop");
  return;
}
else if (imm12 == 0)
{
  ...
}
else
{
  ...
}

Reply:
Thank you for pointing that out. Done